### PR TITLE
Release buffer on exception

### PR DESCRIPTION
**Pull request by jiacheliu3**
### What changes are proposed in this pull request?

This change intends to fix a buffer leak under certain cases

### Why are the changes needed?

Fixes https://github.com/Alluxio/alluxio/issues/13677

### Does this PR introduce any user facing changes?

NA

### CODE REVIEW

File: core/common/src/main/java/alluxio/grpc/DataMessageMarshaller.java [hunk 1]
```diff
@@ -67,8 +67,11 @@ public T parse(InputStream message) {
         message.read(byteBuffer);
         return deserialize(ReadableBuffers.wrap(byteBuffer));
       }
-    } catch (IOException e) {
-      throw new RuntimeException(e);
+    } catch (Throwable t) {
```
Comment (jiacheliu3) [inline core/common/src/main/java/alluxio/grpc/DataMessageMarshaller.java:70]: 2nd thought, we should catch all throwables as we are re-throwing anyways.

**Review by ggezer — APPROVED** [review]

**Comment by ggezer** [comment]
alluxio-bot, merge this please

**Comment by ggezer** [comment]
alluxio-bot, cherry-pick this to branch-2.6 please

**Comment by ggezer** [comment]
alluxio-bot, cherry-pick this to branch-2.5 please

**Comment by ggezer** [comment]
alluxio-bot, cherry-pick this to branch-2.4 please

**Pull request merged**